picoquic/frames.c

```diff
@@ -1017,46 +1017,44 @@ int picoquic_prepare_stream_frame(picoquic_cnx_t* cnx, picoquic_stream_head* str
             if (space < 2 || stream->send_queue == NULL) {
                 length = 0;
             } else {
-                size_t available = (size_t)(stream->send_queue->length - stream->send_queue->offset);
-
-                length = available;
+                length = (size_t)(stream->send_queue->length - stream->send_queue->offset);
+            }
 
-                /* Enforce maxdata per stream on all streams, including stream 0 */
-                if (length >(stream->maxdata_remote - stream->sent_offset)) {
-                    length = (size_t)(stream->maxdata_remote - stream->sent_offset);
-                }
+            /* Enforce maxdata per stream on all streams, including stream 0 */
+            if (length >(stream->maxdata_remote - stream->sent_offset)) {
+                length = (size_t)(stream->maxdata_remote - stream->sent_offset);
+            }
 
-                /* Abide by flow control restrictions, stream 0 is exempt */
-                if (stream->stream_id != 0) {
-                    if (length > (cnx->maxdata_remote - cnx->data_sent)) {
-                        length = (size_t)(cnx->maxdata_remote - cnx->data_sent);
-                    }
+            /* Abide by flow control restrictions, stream 0 is exempt */
+            if (stream->stream_id != 0) {
+                if (length > (cnx->maxdata_remote - cnx->data_sent)) {
+                    length = (size_t)(cnx->maxdata_remote - cnx->data_sent);
                 }
+            }
 
```

Comment:
Should get rid of the stream 0 test, this is a leftover of the pre-draft-13 definition of stream 0

Reply:
Done